src/libasr/codegen/asr_to_cpp.cpp

```diff
@@ -593,6 +593,21 @@ Kokkos::View<T*> from_std_vector(const std::vector<T> &v)
         last_binary_plus = false;
     }
 
+    void visit_ConstantDictionary(const ASR::ConstantDictionary_t &x) {
+        LFORTRAN_ASSERT(x.n_keys == x.n_values);
```

Comment:
I don't think we need the check here since we already do this in AST->ASR.

Reply:
Yes, we don't need to check, but it doesn't hurt either. The assert statements have no overhead in Release mode.